### PR TITLE
Add really simple exit code handling.

### PR DESCRIPTION
**Pull request by pascalholthaus**
**What**: Add really simple exit code handling

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] Conventional commit message
- [ ] Documentation

### CODE REVIEW

File: troubadix/runner.py [hunk 1]
```diff
@@ -231,6 +231,9 @@ def run(
         self._report_info(f"Time elapsed: {datetime.datetime.now() - start}")
         if self.statistic:
             self._report_statistic()
+        # Return true if error exist
+        if len(self.result_counts.result_counts):
+            return True
```
Comment (y0urself) [inline troubadix/runner.py:236]:
```suggestion
        return len(self.result_counts.result_counts) > 0
```

**Review by y0urself — CHANGES_REQUESTED** [review]

File: troubadix/troubadix.py [hunk 2]
```diff
@@ -148,7 +148,9 @@ def main(args=None):
             )
             sys.exit(1)
         info(f"Start linting {len(parsed_args.files)} files ... ")
-        runner.run(parsed_args.files)
+        # Return exit with 1 if error exist
+        if runner.run(parsed_args.files):
+            sys.exit(1)
```
Comment (y0urself) [inline troubadix/troubadix.py:153]:
```suggestion
        sys.exit(!runner.run(parsed_args.files))
```

I think this is fitting?

`sys.exit(True)` returns 1, thus we need to negate it ...

File: troubadix/runner.py (hunk 1 above)
Comment (y0urself) [reply troubadix/runner.py:236]: This is False for 0 and True for everything else ...

File: troubadix/troubadix.py (hunk 2 above)
Comment (bjoernricks) [reply troubadix/troubadix.py:153]: Personally i think it decreases the readability. I would leave it as it is.

File: tests/test_runner.py [hunk 3]
```diff
@@ -109,6 +123,16 @@ def test_runner_run_error(self):
...
         content = nasl_file.read_text(encoding="latin1")
 
+        # Check sys exit 1
+        runner = Runner(
+            n_jobs=1,
+            term=self._term,
+        )
+        with redirect_stdout(io.StringIO()) as _:
+            sys_exit = runner.run([nasl_file])
+        self.assertEqual(sys_exit, True)
```
Comment (y0urself) [inline tests/test_runner.py:133]:
```suggestion
        self.assertTrue(sys_exit)
```
Is the shit! :D

File: tests/test_runner.py [hunk 4]
```diff
@@ -81,6 +81,20 @@ def test_runner_run_ok(self):
...
+            "CheckMissingDescExit",
+        ]
+        runner = Runner(
+            n_jobs=1,
+            included_plugins=included_plugins,
+            term=self._term,
+        )
+        with redirect_stdout(io.StringIO()) as _:
+            sys_exit = runner.run([nasl_file])
+        self.assertEqual(sys_exit, False)
```
Comment (y0urself) [inline tests/test_runner.py:95]:
```suggestion
        self.assertFalse(sys_exit)
```

File: troubadix/troubadix.py (hunk 2 above)
Comment (y0urself) [reply troubadix/troubadix.py:153]: ok

**Review by y0urself — CHANGES_REQUESTED** [review]

File: tests/test_runner.py (hunk 3 above)
Comment (pascalholthaus) [reply tests/test_runner.py]: yes :D

**Review by y0urself — APPROVED** [review]

File: tests/test_runner.py [hunk 5]
```diff
@@ -81,6 +81,20 @@ def test_runner_run_ok(self):
...
+        # Check sys exit 1
+        included_plugins = [
+            "CheckMissingDescExit",
+        ]
+        runner = Runner(
+            n_jobs=1,
+            included_plugins=included_plugins,
+            term=self._term,
+        )
+        with redirect_stdout(io.StringIO()) as _:
```
Comment (bjoernricks) [inline tests/test_runner.py:93]: Sorry i didn't see that initially. Should be just
```python
with redirect_stdout(io.StringIO()):
```

Comment (y0urself) [reply tests/test_runner.py:93]: removed with [8f84c32](https://github.com/greenbone/troubadix/commit/8f84c324bd95ff6029f29258004e44dc9d6ae657)

File: tests/test_runner.py [hunk 6]
```diff
@@ -109,6 +123,20 @@ def test_runner_run_error(self):
...
+        # Check sys exit 1
+        included_plugins = [
+            "CheckCVSSFormat",
+        ]
+        runner = Runner(
+            n_jobs=1,
+            included_plugins=included_plugins,
+            term=self._term,
+        )
+        with redirect_stdout(io.StringIO()) as _:
```
[the remark this answers is not in the document]
Comment (y0urself) [reply tests/test_runner.py:135]: removed with [8f84c32](https://github.com/greenbone/troubadix/commit/8f84c324bd95ff6029f29258004e44dc9d6ae657)

**Pull request merged**